text/0000-min-rust-version.md

```diff
@@ -0,0 +1,179 @@
+- Feature Name: min_rust_version
+- Start Date: 2018-06-28
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add `rust` field to the package section of `Cargo.toml` which will be used to
+specify crate's Minimum Supported Rust Version (MSRV):
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+rust = "1.30"
+```
+
+# Motivation
+[motivation]: #motivation
+
+Currently crates have no way to formally specify MSRV. As a result users can't
+check if crate can be built on their toolchain without building it. It also
+leads to the debate on how to handle crate version change on bumping MSRV,
+conservative approach is to consider such changes as breaking ones, which can
+hinder adoption of new features across ecosystem or result in version number
+inflation, which makes it harder to keep downstream crates up-to-date. More
+relaxed approach on another hand can result in broken crates for user of older
+compiler versions.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+If you target a specific MSRV add a `rust` field to the `[package]` section of
+your `Cargo.toml` with a value equal to the targeted Rust version. If you build
+a crate with a dependency which has MSRV higher than the current version of your
+toolchain, `cargo` will return a compilation error stating the dependency and
+its MSRV. This behavior can be disabled by using `--no-msrv-check` flag.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+During build process (including `run`, `test`, `benchmark`, `verify` and `publish`
+sub-commands) `cargo` will check MSRV requirements of all crates in a dependency
+tree scheduled to be built or checked. Crates which are part of the dependency
+tree, but will not be built are excluded from this check (e.g. target-dependent
+or optional crates).
+
+`rust` field should respect the following minimal requirements:
+- Value should be a version in semver format **without** range operators. Note
+that "1.50" is a valid value and implies "1.50.0".
+- Version can not be bigger than a current stable toolchain (it will be checked
+by crates.io during crate upload).
+- Version can not be smaller than 1.27 (version in which  `package.rust` field
+became a warning instead of an error).
+- Version can not be smaller than release version of a used edition, i.e.
+combination of `rust = "1.27"` and `edition = "2018"` is an invalid one.
+
+# Future work and extensions
+[future-work]: #future-work
```

Comment:
Some things in this section are written in the wrong tense, given that they are matters that this RFC does not decide upon. Because they’re hypothetical or tentative plans, “would” should be used instead of “will”, or else it’s easy to get confused about what’s in scope for implementation, when reading the RFC. For example, part way through reading it I was thinking that influencing version resolution was in scope due to the use of the word “will”, but it is in fact designated as possible future work.

(If you’re not sure what I’m talking about, https://english.stackexchange.com/questions/3657/when-should-i-use-would-would-have-will-and-will-have explains it decently. Also let it be known that I love the term *subjunctive mood*.)

Reply:
Ah, I forgot to fix it after I moved this text to the "future work" section. I will try to fix it.